[Max Autotune][B200] Fix decompose_k test failure

jananisriram opened the pull request:
Summary:
Fix decompose_k test failure (`test_max_autotune_decompose_k `) in `test_max_autotune.py` on B200s by setting `torch._inductor.config` patches for variables `comprehensive_padding` and `shape_padding`. Initial failure was `AssertionError: False is not true : Could not find a split in {3, 9, 2187, 81, 243, 729, 27} in # AOT ID: ['6_forward']`. 

Refactor decompose_k test to follow patch semantics when setting all environment variables within a test.

Test Plan:
`test_max_autotune.py`:
```
buck2 test 'fbcode//mode/opt' fbcode//caffe2/test/inductor:max_autotune -c fbcode.nvcc_arch=b200a -c fbcode.enable_gpu_sections=true -c fbcode.platform010_cuda_version=12.8 -c fbcode.re_gpu_tests=False -- test_max_autotune_decompose_k
```

Differential Revision: D83390563




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben

njriasan commented on test/inductor/test_max_autotune.py:
```diff
@@ -1438,17 +1439,14 @@ def f(x, y):
+        comprehensive_padding=False,
+        shape_padding=False,
+    )
+    @patch_matmul(
```
Any reason you can't just reuse `config.patch` here? This feels like more infrastructure to maintain that isn't reused.

njriasan replied:
Okay makes sense. I don't know enough about this part of the codebase to know if we want to unify eventually, but this seems safer for now.

njriasan reviewed (commented):
See my comment on trying to reuse existing infrastructure.

PaulZhang12 commented on torch/_inductor/template_heuristics/triton.py:
```diff
@@ -315,6 +315,10 @@ def __init__(self) -> None:
             GemmConfig(128, 128, 128, 3, 8),
         ]
 
+        self.blackwell_persistent_addmm_configs: list[BaseConfig] = [
```
Can we add a todo that we need to tune these configs more? Are these for the persistent/tma templates? Why can't blackwell reuse the hopper ones?

jananisriram commented:
@pytorchbot merge

jananisriram committed:
[Max Autotune][B200] Fix decompose_k test failure (#164021)

Summary:

Fix decompose_k test failure (`test_max_autotune_decompose_k `) in `test_max_autotune.py` on B200s by setting `torch._inductor.config` patches for variables `comprehensive_padding` and `shape_padding`. Initial failure was `AssertionError: False is not true : Could not find a split in {3, 9, 2187, 81, 243, 729, 27} in # AOT ID: ['6_forward']`. 

Refactor decompose_k test to follow patch semantics when setting all environment variables within a test.

Test Plan:
`test_max_autotune.py`:
```
buck2 test 'fbcode//mode/opt' fbcode//caffe2/test/inductor:max_autotune -c fbcode.nvcc_arch=b200a -c fbcode.enable_gpu_sections=true -c fbcode.platform010_cuda_version=12.8 -c fbcode.re_gpu_tests=False -- test_max_autotune_decompose_k
```

Reviewed By: eellison, njriasan

Differential Revision: D83390563

jananisriram commented:
@pytorchbot merge
